inskinesis/inskinesis.go

```diff
@@ -201,48 +205,43 @@ func (s *stream) startBatchStreaming() {
 					<-concurrentLimiter
 				}()
 
-				s.totalCount += len(batch)
-				failedCount, err := s.PutRecords(batch)
-				if err != nil {
-					fmt.Printf("Error sending records to Kinesis stream %s: %v\n", s.name, err)
-					s.errChannel <- err
-					return
-				}
-
-				fmt.Printf("Sent %d records to Kinesis stream %s\n", len(batch), s.name)
-
-				s.failedCount += failedCount
+				s.putSingleBatch(batch)
 			}()
 		case <-s.stopBatchChannel:
-			if len(s.logBuffer) == 0 {
-				s.wgBatchChan.Done()
-				return
-			}
-
-			lastBatch := s.logBuffer
-			s.totalCount += len(lastBatch)
-			s.logBuffer = make([]interface{}, 0)
-
-			batches, _ := CreateBatches(lastBatch, s.maxStreamBatchSize, s.maxStreamBatchByteSize)
+			go func() {
+				defer s.wgBatchChan.Done()
 
-			for _, b := range batches {
-				failedCount, err := s.PutRecords(b)
-				s.failedCount += failedCount
-				if err != nil {
-					fmt.Printf("Error sending records to Kinesis stream %s: %v\n", s.name, err)
-					s.errChannel <- err
-					s.wgBatchChan.Done()
+				if len(s.logBuffer) == 0 {
 					return
 				}
 
-			}
+				lastBatch := s.logBuffer
+				s.logBuffer = make([]interface{}, 0)
 
-			s.wgBatchChan.Done()
-			return
+				batches, _ := createBatches(lastBatch, s.maxStreamBatchSize, s.maxStreamBatchByteSize)
+
+				for _, b := range batches {
+					s.putSingleBatch(b)
+				}
```

Comment:
is is working synchronously, expecting to be implemented using the goroutines

Reply:
fixed it